pulp_rpm/app/models/package.py

```diff
@@ -390,3 +394,14 @@ def list_to_createrepo_c(lst):
         package.url = getattr(self, PULP_PACKAGE_ATTRS.URL)
         package.version = getattr(self, PULP_PACKAGE_ATTRS.VERSION)
         return package
+
+
+@receiver(pre_save, sender=Package)
```

Comment:
Consider using [django-lifecycle](https://github.com/rsinger86/django-lifecycle) instead of a signal. There's an example in ggainey's PR: https://github.com/pulp/pulpcore/pull/1027

Reply:
A unit test for this would be awesome too.